### PR TITLE
:building_construction: :construction:   Database performance audit

### PR DESCRIPTION
**Pull request by SimonGideon**
## Database performance audit.
- Improved execution time for each query.
- Applied indexing

|No.| Query :recycle:   | Before Indexing(ms) :slightly_frowning_face:  | After Indexing(ms) :slightly_smiling_face:  |
|---| ------------- | ------------- | ------------- |
|1.| SELECT COUNT(*) FROM visits where animal_id = 4; | 174.078 |57.323 |
|2.|SELECT * FROM visits where vet_id = 2; | 321.501  |314.616  |
|3.| SELECT * FROM owners where email = 'owner_18327@mail.com' | 155.555  | 21.833  |

Screenshot from 2023-02-07 12-24-19
Screenshot from 2023-02-07 12-23-34
Screenshot from 2023-02-07 12-22-58
Screenshot from 2023-02-07 11-25-35
Screenshot from 2023-02-07 11-25-11
Screenshot from 2023-02-07 02-50-54
Screenshot from 2023-02-07 02-52-14
Screenshot from 2023-02-07 02-51-35
Screenshot from 2023-02-07 02-46-24

### CODE REVIEW

**Review by cynthiainga — COMMENTED** [review]
Hi @SimonGideon, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn’t have your partner's contribution. This is a pair-programming activity you should work on with your partner, and if you have some issues with your partner kindly contact student success 🙏 .

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: data.sql [hunk 1]
```diff
@@ -154,3 +154,19 @@ INSERT INTO visits (animal_id, vet_id, visit_date)
...
+		 
+-- This will add 3.594.280 visits considering you have 10 animals, 4 vets, and it will use around ~87.000 timestamps (~4min approx.)
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
+
+-- This will add 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
+insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
+
+-- Creating an INDEX
+CREATE INDEX visits(animal_vet_id_idx) ON visits(animal_id, vet_id);
+CREATE INDEX email_idx ON owners(email);
```
Comment (thecodechaser) [inline data.sql:172]: - [ ] Please move these lines of code into the `schema.sql` file as these are the schema-related code.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @SimonGideon, and @enoqueJonas👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (SimonGideon) [reply data.sql:172]: :slightly_smiling_face: Thanks @thecodechaser for the change requested. Implemented on the future commits.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @SimonGideon, and @enoqueJonas👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by prolajumokeoni** [comment]
Hi @,Team

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a wrong pull request for this milestone: Create a database based on a diagram

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**